src/inmanta/moduletool.py

```diff
@@ -613,3 +633,93 @@ def freeze(self, outfile, recursive, operator, module=None):
         finally:
             if close:
                 outfile.close()
+
+
+class ModuleBuildFailedError(Exception):
+    def __init__(self, msg: str, *args, **kwargs) -> None:
+        self.msg = msg
+        super(ModuleBuildFailedError, self).__init__(*args, **kwargs)
+
+    def __str__(self) -> str:
+        return self.msg
+
+
+class V2ModuleBuilder:
+    def __init__(self, module_path: str) -> None:
+        """
+        :raises InvalidModuleException: The given module_path doesn't reference a valid module.
+        :raises ModuleBuildFailedError: Module build was unsuccessful.
+        """
+        self._module = ModuleV2(project=None, path=os.path.abspath(module_path))
+
+    def build(self) -> None:
+        output_directory = os.path.join(self._module.path, "dist")
```

Comment:
I think it would be good to make the output directory configurable. This way `inmanta module install` can build to a temp dir. I could just as well make the change a part of my pull request if you prefer.

Reply:
Good idea. I will implement it.